src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -13,9 +14,31 @@
 
 from .base import AgentSchedulingComponent
 
+import cProfile
+import inspect
+import threading as mt
 
-# ==============================================================================
-#
+cprof = cProfile.Profile()
+
+def cprof_it(func):
+    def wrapper(*args, **kwargs):
+            retval = cprof.runcall(func, *args, **kwargs)
+            return retval
+
+    return wrapper
+
+def dec_all_methods(dec):
+    def dectheclass(cls):
+        self_thread = mt.current_thread()
+        if self_thread.name == 'MainThread' and \
+                "CONTINUOUS" in os.getenv("RADICAL_PILOT_CPROFILE_COMPONENTS", "").split():
+            for name, m in inspect.getmembers(cls, inspect.ismethod):
+                setattr(cls, name, dec(m))
+        return cls
+    return dectheclass
+
+#==============================================================================
```

Comment:
Out of interest: why is cprof actually useful on the scheduler?  Aren't the critical parts mostly in python land?